sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/aggregate/interfaces.scala

```diff
@@ -95,12 +95,32 @@ private[sql] case class AggregateExpression2(
   override def toString: String = s"(${aggregateFunction},mode=$mode,isDistinct=$isDistinct)"
 }
 
-abstract class AggregateFunction2
-  extends Expression with ImplicitCastInputTypes {
+sealed abstract class AggregateFunction2 extends Expression with ImplicitCastInputTypes {
```

Comment:
I plan to write some proper Scaladoc for this class to explain the role of its methods.